lighthouse-core/test/lib/tracehouse/trace-processor-test.js

```diff
@@ -322,6 +324,31 @@ describe('TraceProcessor', () => {
       });
     });
 
+    describe('finds correct LCP', () => {
+      it('if there was a tracingStartedInPage after the frame\'s navStart', () => {
+        const trace = TraceProcessor.computeTraceOfTab(lcpTrace);
+        assert.equal(trace.mainFrameIds.frameId, '906A10385298DD996B521026AF4DA204');
+        assert.equal(trace.navigationStartEvt.ts, 1671221915754);
+        assert.equal(trace.firstContentfulPaintEvt.ts, 1671226617803);
+        assert.equal(trace.largestContentfulPaintEvt.ts, 1671236939268);
+        assert.ok(!trace.invalidLcp);
+      });
+
+      it('invalidates if last event is ::Invalidate', () => {
+        const invalidLcpTrace = createTestTrace({navigationStart: 0, traceEnd: 2000});
```

Comment:
> create-test-trace.js

This doesn't have the capabilities I think ya'll expected it to have. Should I attempt to parameterize adding frame events to `createTestTrace`?

Reply:
> This doesn't have the capabilities I think ya'll expected it to have. Should I attempt to parameterize adding frame events to `createTestTrace`?

we should give it some helpful utilities at some point, but that's shouldn't be necessary for this testing. The main purpose of the file right now is just that it gives a minimal processable trace without having to copy and paste one everywhere.

Pushing new LCP events with the same frame as navStart (or the real deal in `TracingStartedInBrowser` but that's more work) should be sufficient for a test?

Reply:
cool. should be good now then.